backend/src/services/super-admin/super-admin-service.ts

```diff
@@ -87,17 +87,24 @@ export const superAdminServiceFactory = ({
 
     // reset on initialized
     await keyStore.deleteItem(ADMIN_CONFIG_KEY);
-    const serverCfg = await serverCfgDAL.findById(ADMIN_CONFIG_DB_UUID);
-    if (serverCfg) return;
-
-    const newCfg = await serverCfgDAL.create({
-      // @ts-expect-error id is kept as fixed for idempotence and to avoid race condition
-      id: ADMIN_CONFIG_DB_UUID,
-      initialized: false,
-      allowSignUp: true,
-      defaultAuthOrgId: null
+    const serverCfg = await serverCfgDAL.transaction(async (tx) => {
+      await tx.raw("SELECT pg_advisory_xact_lock(?)", [PgSqlLock.SuperAdminInit]);
+      const serverCfgInDB = await serverCfgDAL.findById(ADMIN_CONFIG_DB_UUID, tx);
```

Comment:
because you are already inside the txn, you won't see new data anymore if it was created by some other instance